Gems/EMotionFX/Code/EMotionFX/Pipeline/SceneAPIExt/Rules/RootMotionExtractionRule.cpp

```diff
@@ -15,8 +15,12 @@
 
 namespace EMotionFX::Pipeline::Rule
 {
-    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-    RootMotionExtractionRule::RootMotionExtractionRule(const RootMotionExtractionData& data)
+    RootMotionExtractionRule::RootMotionExtractionRule()
+    {
+        m_data = AZStd::make_shared<RootMotionExtractionData>();
+    }
+
+    RootMotionExtractionRule::RootMotionExtractionRule(const AZStd::shared_ptr<RootMotionExtractionData>& data)
```

Comment:
suggestion: To show we are taking ownership of the pointer we might want to instead pass this by value and then `AZStd::move` the variable `data` into `m_data`.

```
RootMotionExtractionRule::RootMotionExtractionRule(AZStd::shared_ptr<RootMotionExtractionData> data)
    : m_data(AZStd::move(data))
{
    ...
```